### PR TITLE
Add setting output_format_parquet_compliant_nested_types to produce more compatible Parquet files

### PR DESCRIPTION
**Pull request by al13n321**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Fixed a reported incompatibility of the Parquet files written by ClickHouse with Google BigQuery: https://github.com/ClickHouse/ClickHouse/issues/49997

See [the issue](https://github.com/ClickHouse/ClickHouse/issues/49997) for explanation.

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
This is an automated comment for commit d7107a749a9f0407b33bda805384fe42aa114c08 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50001/d7107a749a9f0407b33bda805384fe42aa114c08/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by al13n321** [comment]
Tested manually:
```
INSERT INTO FUNCTION file('ds/t.parquet')
SETTINGS engine_file_truncate_on_insert = 1, output_format_parquet_compliant_nested_types = 0
SELECT [1, 2, 3] AS a
SETTINGS engine_file_truncate_on_insert = 1, output_format_parquet_compliant_nested_types = 0

SELECT c
FROM file('ds/t.parquet', ParquetMetadata)
ARRAY JOIN columns AS c

┌─c───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ ('item','a.list.item',2,1,'INT32','Int(bitWidth=8, isSigned=false)','LZ4',75,78,'-4%',['RLE_DICTIONARY','PLAIN','RLE']) │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘


INSERT INTO FUNCTION file('ds/t.parquet')
SETTINGS engine_file_truncate_on_insert = 1, output_format_parquet_compliant_nested_types = 1
SELECT [1, 2, 3] AS a
SETTINGS engine_file_truncate_on_insert = 1, output_format_parquet_compliant_nested_types = 1

SELECT c
FROM file('ds/t.parquet', ParquetMetadata)
ARRAY JOIN columns AS c

┌─c─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ ('element','a.list.element',2,1,'INT32','Int(bitWidth=8, isSigned=false)','LZ4',75,78,'-4%',['RLE_DICTIONARY','PLAIN','RLE']) │
└───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

File: src/Core/Settings.h [hunk 1]
```diff
@@ -913,6 +913,7 @@ class IColumn;
     M(Bool, output_format_parquet_fixed_string_as_fixed_byte_array, true, "Use Parquet FIXED_LENGTH_BYTE_ARRAY type instead of Binary for FixedString columns.", 0) \
     M(ParquetVersion, output_format_parquet_version, "2.latest", "Parquet format version for output format. Supported versions: 1.0, 2.4, 2.6 and 2.latest (default)", 0) \
     M(ParquetCompression, output_format_parquet_compression_method, "lz4", "Compression method for Parquet output format. Supported codecs: snappy, lz4, brotli, zstd, gzip, none (uncompressed)", 0) \
+    M(Bool, output_format_parquet_compliant_nested_types, true, "In parquet file schema, use name 'element' instead of 'item' for list elements. This is a historical artifact of Arrow library implementation. Generally increases compatibility, except perhaps with some old versions of Arrow.", 0) \
```
Comment (Avogar) [inline src/Core/Settings.h:916]: Let's add it to `SettingsChangesHistory`?

**Review by Avogar — APPROVED** [review]

**Comment by Avogar** [comment]
`Stress test (ubsan) ` - https://github.com/ClickHouse/ClickHouse/issues/49553

**Pull request merged**